src/installer.ts

```diff
@@ -1,223 +1,114 @@
+  files: IGoVersionFile[];
 }
 
-function setGoEnvironmentVariables(goRoot: string) {
-  core.exportVariable('GOROOT', goRoot);
+export async function findMatch(
+  versionSpec: string,
+  stable: boolean
+): Promise<IGoVersion | undefined> {
+  let archFilter = sys.getArch();
+  let platFilter = sys.getPlatform();
 
-  const goPath: string = process.env['GOPATH'] || '';
-  const goBin: string = process.env['GOBIN'] || '';
+  let result: IGoVersion | undefined;
+  let match: IGoVersion | undefined;
 
-  // set GOPATH and GOBIN as user value
-  if (goPath) {
-    core.exportVariable('GOPATH', goPath);
+  const dlUrl: string = 'https://golang.org/dl/?mode=json&include=all';
+  let candidates: IGoVersion[] | null = await module.exports.getVersions(dlUrl);
+  if (!candidates) {
+    throw new Error(`golang download url did not return results`);
   }
-  if (goBin) {
-    core.exportVariable('GOBIN', goBin);
-  }
-}
 
-// This function is required to convert the version 1.10 to 1.10.0.
-// Because caching utility accept only sementic version,
-// which have patch number as well.
-function normalizeVersion(version: string): string {
-  const versionPart = version.split('.');
-  if (versionPart[1] == null) {
-    //append minor and patch version if not available
-    return version.concat('.0.0');
-  } else {
-    // handle beta and rc: 1.10beta1 => 1.10.0-beta1, 1.10rc1 => 1.10.0-rc1
-    if (versionPart[1].includes('beta') || versionPart[1].includes('rc')) {
-      versionPart[1] = versionPart[1]
-        .replace('beta', '.0-beta')
-        .replace('rc', '.0-rc');
-      return versionPart.join('.');
-    }
-  }
+  let goFile: IGoVersionFile | undefined;
+  for (let i = 0; i < candidates.length; i++) {
+    let candidate: IGoVersion = candidates[i];
+    let version = candidate.version.replace('go', '');
 
-  if (versionPart[2] == null) {
-    //append patch version if not available
-    return version.concat('.0');
-  } else {
-    // handle beta and rc: 1.8.5beta1 => 1.8.5-beta1, 1.8.5rc1 => 1.8.5-rc1
-    if (versionPart[2].includes('beta') || versionPart[2].includes('rc')) {
-      versionPart[2] = versionPart[2]
-        .replace('beta', '-beta')
-        .replace('rc', '-rc');
-      return versionPart.join('.');
+    // 1.13.0 is advertised as 1.13 preventing being able to match exactly 1.13.0
+    // since a semver of 1.13 would match latest 1.13
+    let parts: string[] = version.split('.');
+    if (parts.length == 2) {
+      version = version + '.0';
     }
-  }
-
-  return version;
-}
 
-async function determineVersion(version: string): Promise<string> {
-  if (!version.endsWith('.x')) {
-    const versionPart = version.split('.');
-
-    if (versionPart[1] == null || versionPart[2] == null) {
-      return await getLatestVersion(version.concat('.x'));
-    } else {
-      return version;
+    debug(`check ${version} satisfies ${versionSpec}`);
+    if (semver.satisfies(version, versionSpec) && candidate.stable == stable) {
```

Comment:
should this be `if (semver.satisfies(version, versionSpec) && (!stable || candidate.stable === stable))`

Reply:
the description for stable is: `Whether to download only stable versions`

so i would expect if a customer specifies `stable: false`, it would be OK if they get a stable version

Reply:
Good catch.  I also need to add a test for that.  Yep, the expectation is that if they say false, then it gets the latest match on that semver spec regardless of stable.  In other words, true means we look at the stable attribute, false means we don't even consider the candidate.stable data.